spec/mobility/plugins/active_record/query_spec.rb

```diff
@@ -153,4 +153,44 @@
       end).to eq([article_ja])
     end
   end
+
+  describe "regression for #513" do
+    plugins :active_record, :query
+    before do
+      m = ActiveRecord::Migration.new
+      m.verbose = false
+
+      m.create_table :cars
+      stub_const('Car', Class.new(ActiveRecord::Base) do
+        has_many :parking_lots
+        has_many :car_parts
+        scope :some_car_scope, -> { i18n }
+      end)
+      translates Car, backend: :column
+
+      m.create_table(:parking_lots) { |t| t.integer :car_id }
+      stub_const('ParkingLot', Class.new(ActiveRecord::Base) do
+        belongs_to :car
+      end)
+
+      m.create_table(:car_parts) { |t| t.integer :car_id }
+      stub_const('CarPart', Class.new(ActiveRecord::Base) do
+        belongs_to :car
+      end)
+      translates CarPart, backend: :column
+    end
+    after do
+      m = ActiveRecord::Migration.new
+      m.verbose = false
+      m.drop_table :cars
+      m.drop_table :parking_lots
+      m.drop_table :car_parts
+    end
+
+    it "does not raise NameError" do
+      query = ParkingLot.includes(car: :car_parts).references(:car).merge(Car.some_car_scope)
+      expect { query.first }.not_to raise_error
+      expect { query.order(:car_id) }.not_to raise_error
```

Comment:
@DJA89 I managed to narrow the problem down a bit, it doesn't even require any actual translated attributes, nor the `through` relation, and only `Car` needs to extend Mobility.

I'm still not really clear what the underlying problem is but since it no longer occurs in Rails 7.0 (master) I'm assuming it's something that's been "fixed", so this will be temporary until Mobility no longer supports versions < Rails 7 (still a long way away!).